lib/cisco_node_utils/span_session.rb

```diff
@@ -0,0 +1,125 @@
+# Insert appropriate license here
+require_relative 'node_util'
+require_relative 'interface'
+
+
+module Cisco
+  # node utils class for SPAN (switchport analyzer) sessions
+  class SpanSession < NodeUtil
+    attr_reader :session_id
+
+    def initialize(session_id)
+      validate_args(session_id)
```

Comment:
Better to cast `session_id` as an integer when you pass it into `validate`.

`validate_args(session_id.to_i)`